docs/developing-component.md

```diff
@@ -0,0 +1,88 @@
+| Bindings | [components-contrib/bindings](https://github.com/dapr/components-contrib/tree/master/bindings) | [Kafka](https://github.com/dapr/components-contrib/tree/master/bindings/kafka) | [concept](https://github.com/dapr/docs/tree/master/concepts/bindings), [input howto](https://github.com/dapr/docs/tree/master/howto/trigger-app-with-input-binding), [output howto](https://github.com/dapr/docs/tree/master/howto/send-events-with-output-bindings), [api spec](https://github.com/dapr/docs/blob/master/reference/api/bindings.md) |
+| Secret Store | [components-contrib/secretstore](https://github.com/dapr/components-contrib/tree/master/secretstores) | [Kubernetes](https://github.com/dapr/components-contrib/tree/master/secretstores/kubernetes), [Azure Keyvault](https://github.com/dapr/components-contrib/tree/master/secretstores/azure/keyvault) | [concept](https://github.com/dapr/docs/blob/master/concepts/components/secrets.md), [howto](https://github.com/dapr/docs/tree/master/howto/setup-secret-store)|
+| Middleware | [components-contrib/middleware](https://github.com/dapr/components-contrib/tree/master/middleware) | [Oauth2](https://github.com/dapr/components-contrib/blob/master/middleware/http/oauth2/oauth2_middleware.go) | [concept](https://github.com/dapr/docs/blob/master/concepts/middleware/middleware.md), [howto](https://github.com/dapr/docs/tree/master/howto/authorization-with-oauth) |
+| Exporter | [components-contrib/exporters](https://github.com/dapr/components-contrib/tree/master/exporters) | [Zipkin](https://github.com/dapr/components-contrib/blob/master/exporters/zipkin/zipkin_exporter.go) | [concept](https://github.com/dapr/docs/tree/master/concepts/distributed-tracing), [howto](https://github.com/dapr/docs/tree/master/howto/diagnose-with-tracing) |
+| Service Discovery | [components-contrib/servicediscovery](https://github.com/dapr/components-contrib/tree/master/servicediscovery) | [mdns](https://github.com/dapr/components-contrib/blob/master/servicediscovery/mdns/mdns.go) | [howto](https://github.com/dapr/docs/tree/master/howto/invoke-and-discover-services) |
+
+### Running unit-test
+
+```bash
+make test
+```
+
+### Running linting
+
+```bash
+make lint
+```
+
+## Validating with Dapr core
+
+1. Make sure you clone dapr and component-contrib repos under $GOPATH/src/github.com/dapr
+
+2. Replace github.com/dapr/components-contrib reference to the local component-contrib
+
+```bash
+go mod edit -replace github.com/dapr/component-contrib=../component-contrib
+```
+
+1. Import your component to dapr [main.go](https://github.com/dapr/dapr/blob/d17e9243b308e830649b0bf3af5f6e84fd543baf/cmd/daprd/main.go#L79)
+
+2. Register your component in dapr [main.go](https://github.com/dapr/dapr/blob/d17e9243b308e830649b0bf3af5f6e84fd543baf/cmd/daprd/main.go#L153-L226)(e.g. binding)
+
+3. Build dapr code locally
+
+```bash
+make DEBUG=1 build
+```
+
+6. Replace the installed daprd with the test binary (then dapr cli will use the test binary)
```

Comment:
might want to suggest people backup their existing `daprd` bin first in case they want to switch back easily without a reinstall/rebuild i.e.
` /usr/local/bin/daprd /usr/local/bin/daprd.bak && cp ./dist/linux_amd64/debug/daprd /usr/local/bin`

Reply:
Fixed.